cmd/mdatagen/internal/loader.go

```diff
@@ -373,3 +401,50 @@ func packageName() (string, error) {
 	}
 	return strings.TrimSpace(string(output)), nil
 }
+
+type featureGate struct {
+	// Required.
+	Id string `mapstructure:"id"`
+	// Description describes the purpose of the attribute.
+	Description string `mapstructure:"description"`
+	// Stage current stage at which the feature gate is in the development lifecyle
+	Stage string `mapstructure:"stage"`
+	// ReferenceURL can optionally give the url of the feature_gate
+	ReferenceURL string `mapstructure:"reference_url"`
+	// FromVersion optional field which gives the release version from which the gate has been given the current stage
+	FromVersion string `mapstructure:"from_version"`
+	// ToVersion optional field which gives the release version till which the gate the gate had the given lifecycle stage
+	ToVersion string `mapstructure:"to_version"`
+	// FeatureGateName name of the feature gate
+	FeatureGateName featureGateName `mapstructure:"-"`
+}
+
+func (f *featureGate) validate(parser *confmap.Conf) error {
+	if !parser.IsSet("id") {
```

Comment:
We should also check for stage, that is also a required field

Reply:
done